modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/QueryTaskExecutorImpl.java

```diff
@@ -82,16 +85,18 @@ public void exceptionHandler(Thread.UncaughtExceptionHandler eHnd) {
     @Override public void onStart(GridKernalContext ctx) {
         exceptionHandler(ctx.uncaughtExceptionHandler());
 
-        CalciteQueryProcessor proc = Objects.requireNonNull(Commons.lookupComponent(ctx, CalciteQueryProcessor.class));
-
-        stripedThreadPoolExecutor(new IgniteStripedThreadPoolExecutor(
+        IgniteStripedThreadPoolExecutor executor = new IgniteStripedThreadPoolExecutor(
             ctx.config().getQueryThreadPoolSize(),
             ctx.igniteInstanceName(),
             "calciteQry",
             this,
             false,
             0
-        ));
+        );
+
+        stripedThreadPoolExecutor(executor);
+
+        executor.registerMetrics(ctx.metric().registry(metricName(THREAD_POOLS, THREAD_POOL_NAME)));
```

Comment:
Should we also add a view like for DataStreamer pool? 
`        ctx.systemView().registerInnerCollectionView("CalciteQueryPool", "Pool for query executions", ...`

Reply:
We don't have views over `IgniteStripedThreadPoolExecutor` now, only for `StripedExecutor`, I think we can add views for all these pools in the new ticket